[BUG] WSS connection fails

## Description

Using JS client Socket.IO v4.8.3
`const socket = io('https://myserver:8085/terminal', { transports: ["websocket"] });`
WSS connection fails with
No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
With 4.0.0-alpha commit fddd5d1 was OK

## Steps to Reproduce

1. connect using a JS browser client 

## Expected Behavior

Connection successful

## Actual Behavior

Connection failed

## Environment

- **Java Version**: 17
- **OS**: Windows
- **Netty-SocketIO Version**: 4.0.0-beta
- **Module**: core/spring

## Code Example

```javascript
const socket = io('https://myserver:8085/terminal', { transports: ["websocket"] });
```

## Logs/Stack Trace

```
javax.net.ssl|ERROR|C5|multiThreadIoEventLoopGroup-3-35|2026-04-08 10:28:25.815 CEST|TransportContext.java:375|Fatal (INTERNAL_ERROR): problem unwrapping net record (
"throwable" : {
  javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
  	at java.base/sun.security.ssl.HandshakeContext.<init>(HandshakeContext.java:173)
  	at java.base/sun.security.ssl.ServerHandshakeContext.<init>(ServerHandshakeContext.java:63)
  	at java.base/sun.security.ssl.TransportContext.kickstart(TransportContext.java:253)
  	at java.base/sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:546)
  	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:506)
  	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:482)
  	at java.base/javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679)
  	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:307)
  	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1507)
  	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1398)
  	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1449)
  	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:545)
  	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:484)
  	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356)
  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1429)
  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918)
  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:172)
  	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.handle(AbstractNioChannel.java:445)
  	at io.netty.channel.nio.NioIoHandler$DefaultNioRegistration.handle(NioIoHandler.java:388)
  	at io.netty.channel.nio.NioIoHandler.processSelectedKey(NioIoHandler.java:596)
  	at io.netty.channel.nio.NioIoHandler.processSelectedKeysOptimized(NioIoHandler.java:571)
  	at io.netty.channel.nio.NioIoHandler.processSelectedKeys(NioIoHandler.java:512)
  	at io.netty.channel.nio.NioIoHandler.run(NioIoHandler.java:484)
  	at io.netty.channel.SingleThreadIoEventLoop.runIo(SingleThreadIoEventLoop.java:225)
  	at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:196)
  	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1195)
  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
  	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
  	at java.base/java.lang.Thread.run(Thread.java:840)}

)
javax.net.ssl|ALL|C5|multiThreadIoEventLoopGroup-3-35|2026-04-08 10:28:25.815 CEST|SSLSessionImpl.java:1192|Invalidated session:  Session(1775636905812|SSL_NULL_WITH_NULL_NULL)
2026-04-08 10:28:25 ERROR com.socketio4j.socketio.listener.DefaultExceptionListener - javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
io.netty.handler.codec.DecoderException: javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:515)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1429)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:172)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.handle(AbstractNioChannel.java:445)
	at io.netty.channel.nio.NioIoHandler$DefaultNioRegistration.handle(NioIoHandler.java:388)
	at io.netty.channel.nio.NioIoHandler.processSelectedKey(NioIoHandler.java:596)
	at io.netty.channel.nio.NioIoHandler.processSelectedKeysOptimized(NioIoHandler.java:571)
	at io.netty.channel.nio.NioIoHandler.processSelectedKeys(NioIoHandler.java:512)
	at io.netty.channel.nio.NioIoHandler.run(NioIoHandler.java:484)
	at io.netty.channel.SingleThreadIoEventLoop.runIo(SingleThreadIoEventLoop.java:225)
	at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:196)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1195)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
	at java.base/sun.security.ssl.HandshakeContext.<init>(HandshakeContext.java:173)
	at java.base/sun.security.ssl.ServerHandshakeContext.<init>(ServerHandshakeContext.java:63)
	at java.base/sun.security.ssl.TransportContext.kickstart(TransportContext.java:253)
	at java.base/sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:546)
	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:506)
	at java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:482)
	at java.base/javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679)
	at io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:307)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1507)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1398)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1449)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:545)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:484)
	... 17 common frames omitted
```

## Additional Context

WSS is used
# HTTP/HTTPS
server.port=8443

# socket.io
netty-socket-io.port=8085
netty-socket-io.need-client-auth=false
netty-socket-io.enable-cors=true
netty-socket-io.allow-headers=*
netty-socket-io.origin=*

netty-socket-io.ssl.key-store=file:C:/mystore.p12
netty-socket-io.ssl.key-store-type=PKCS12
netty-socket-io.ssl.key-store-password=password
netty-socket-io.ssl.key-alias=myalias
netty-socket-io.ssl.trust-store=file:C:/mystore.p12
netty-socket-io.ssl.trust-store-type=PKCS12
netty-socket-io.ssl.trust-store-password=password
netty-socket-io.ssl.trust-alias=myalias

--- a/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketIOChannelInitializer.java
+++ b/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketIOChannelInitializer.java
@@ -226,7 +226,7 @@ public class SocketIOChannelInitializer extends ChannelInitializer<Channel> impl
 
         SslContextBuilder builder = SslContextBuilder.forServer(kmf).sslProvider(sslProvider);
         String sslProtocol = socketSslConfig.getSSLProtocol();
-        if (socketSslConfig.isSslProtocolExplicitlySet() && sslProtocol != null) {
+        if (socketSslConfig.isSSLProtocolExplicitlySet() && sslProtocol != null) {
             // SocketSslConfig historically accepted SSLContext algorithm names like "TLS".
             // SslContextBuilder.protocols(...) expects concrete enabled protocol versions.
             if (isTlsProtocolVersion(sslProtocol)) {

--- a/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketIOChannelInitializer.java
+++ b/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketIOChannelInitializer.java
@@ -226,7 +226,7 @@ public class SocketIOChannelInitializer extends ChannelInitializer<Channel> impl
 
         SslContextBuilder builder = SslContextBuilder.forServer(kmf).sslProvider(sslProvider);
         String sslProtocol = socketSslConfig.getSSLProtocol();
-        if (sslProtocol != null) {
+        if (socketSslConfig.isSslProtocolExplicitlySet() && sslProtocol != null) {
             // SocketSslConfig historically accepted SSLContext algorithm names like "TLS".
             // SslContextBuilder.protocols(...) expects concrete enabled protocol versions.
             if (isTlsProtocolVersion(sslProtocol)) {

--- a/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketSslConfig.java
+++ b/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketSslConfig.java
@@ -201,11 +201,14 @@ public class SocketSslConfig {
     /**
      * Whether {@link #setSSLProtocol(String)} has been called.
      * <p>
-     * This allows the server to distinguish between a default value and an explicit configuration,
-     * so modern JSSE defaults (TLSv1.2/1.3) can be used when the user didn't request a specific version.
+     * This flag is driven purely by invocation, not by the semantic intent of the value. Integrations that
+     * populate {@link SocketSslConfig} programmatically (e.g., factories, recorders, property binders)
+     * should avoid calling {@link #setSSLProtocol(String)} unless they intentionally want to pin the enabled
+     * TLS protocol version via {@code SslContextBuilder.protocols(...)}. When not explicitly set, the server
+     * will rely on the provider defaults (typically TLSv1.2/1.3 on modern JSSE).
      * </p>
      */
-    public boolean isSslProtocolExplicitlySet() {
+    public boolean isSSLProtocolExplicitlySet() {
         return sslProtocolExplicitlySet;
     }
 }

--- a/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketSslConfig.java
+++ b/netty-socketio-core/src/main/java/com/socketio4j/socketio/SocketSslConfig.java
@@ -24,6 +24,7 @@ import javax.net.ssl.KeyManagerFactory;
 
 public class SocketSslConfig {
     private String sslProtocol = "TLSv1";
+    private boolean sslProtocolExplicitlySet;
 
     private String keyStoreFormat = "JKS";
     private InputStream keyStore;
@@ -190,9 +191,21 @@ public class SocketSslConfig {
      */
     public void setSSLProtocol(String sslProtocol) {
         this.sslProtocol = sslProtocol;
+        this.sslProtocolExplicitlySet = true;
     }
 
     public String getSSLProtocol() {
         return sslProtocol;
+    }
+
+    /**
+     * Whether {@link #setSSLProtocol(String)} has been called.
+     * <p>
+     * This allows the server to distinguish between a default value and an explicit configuration,
+     * so modern JSSE defaults (TLSv1.2/1.3) can be used when the user didn't request a specific version.
+     * </p>
+     */
+    public boolean isSslProtocolExplicitlySet() {
+        return sslProtocolExplicitlySet;
     }
 }

--- a/netty-socketio-core/src/test/java/com/socketio4j/socketio/transport/SocketIoJavaClientSslTest.java
+++ b/netty-socketio-core/src/test/java/com/socketio4j/socketio/transport/SocketIoJavaClientSslTest.java
@@ -142,6 +142,85 @@ public class SocketIoJavaClientSslTest {
         }
     }
 
+    @Test
+    public void shouldHandshakeOverWssWhenSslProtocolNotExplicitlyConfigured() throws Exception {
+        CountDownLatch serverReceivedHello = new CountDownLatch(1);
+        CountDownLatch clientReceivedAck = new CountDownLatch(1);
+        CountDownLatch engineHandshakeDone = new CountDownLatch(1);
+        AtomicReference<Throwable> connectError = new AtomicReference<>();
+
+        server = startServer(0, testSslConfigWithoutExplicitProtocol(), serverReceivedHello);
+        int port = awaitBoundPort(server);
+        assertTrue(port > 0, "server did not bind an ephemeral port");
+
+        X509TrustManager trustAll = new X509TrustManager() {
+            @Override
+            public void checkClientTrusted(X509Certificate[] chain, String authType) {
+            }
+
+            @Override
+            public void checkServerTrusted(X509Certificate[] chain, String authType) {
+            }
+
+            @Override
+            public X509Certificate[] getAcceptedIssuers() {
+                return new X509Certificate[0];
+            }
+        };
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, new TrustManager[] { trustAll }, new SecureRandom());
+
+        OkHttpClient okHttp = new OkHttpClient.Builder()
+                .sslSocketFactory(sslContext.getSocketFactory(), trustAll)
+                .hostnameVerifier((hostname, session) -> true)
+                .readTimeout(1, TimeUnit.MINUTES)
+                .build();
+
+        IO.Options opts = new IO.Options();
+        opts.forceNew = true;
+        opts.reconnection = false;
+        opts.transports = new String[] { "websocket" };
+        opts.webSocketFactory = okHttp;
+        opts.callFactory = okHttp;
+
+        Socket socket = IO.socket("https://127.0.0.1:" + port, opts);
+        try {
+            socket.on(Socket.EVENT_CONNECT_ERROR, args -> {
+                Object first = args.length > 0 ? args[0] : null;
+                if (first instanceof Throwable) {
+                    connectError.set((Throwable) first);
+                } else {
+                    connectError.set(new IllegalStateException(String.valueOf(first)));
+                }
+                engineHandshakeDone.countDown();
+            });
+            socket.on(Socket.EVENT_CONNECT, args -> {
+                try {
+                    JSONObject payload = new JSONObject();
+                    payload.put("a", 1);
+                    socket.emit("hello", payload, (Ack) ackArgs -> {
+                        if (ackArgs.length > 0 && "ok".equals(String.valueOf(ackArgs[0]))) {
+                            clientReceivedAck.countDown();
+                        }
+                    });
+                } catch (Exception e) {
+                    connectError.set(e);
+                } finally {
+                    engineHandshakeDone.countDown();
+                }
+            });
+            socket.connect();
+
+            assertTrue(engineHandshakeDone.await(20, TimeUnit.SECONDS),
+                    () -> "Engine.IO handshake did not complete: " + connectError.get());
+            assertNull(connectError.get(), () -> "connect_error: " + connectError.get());
+            assertTrue(serverReceivedHello.await(15, TimeUnit.SECONDS), "server did not receive hello event");
+            assertTrue(clientReceivedAck.await(15, TimeUnit.SECONDS), "client did not receive ack");
+        } finally {
+            socket.disconnect();
+        }
+    }
+
     /**
      * Exercises polling first (POST body via {@code PollingTransport.onPost}), then upgrade to WebSocket.
      * Server pushes an event right after connect so delivery runs while the client may still be on polling.
@@ -447,6 +526,18 @@ public class SocketIoJavaClientSslTest {
     private SocketSslConfig testSslConfig() throws Exception {
         SocketSslConfig ssl = new SocketSslConfig();
         ssl.setSSLProtocol("TLSv1.2");
+        ssl.setKeyStoreFormat("PKCS12");
+        ssl.setKeyStorePassword("password");
+
+        InputStream ks = SocketIoJavaClientSslTest.class.getClassLoader()
+                .getResourceAsStream("ssl/test-socketio.p12");
+        assertNotNull(ks, "Missing test keystore resource ssl/test-socketio.p12");
+        ssl.setKeyStore(ks);
+        return ssl;
+    }
+
+    private SocketSslConfig testSslConfigWithoutExplicitProtocol() throws Exception {
+        SocketSslConfig ssl = new SocketSslConfig();
         ssl.setKeyStoreFormat("PKCS12");
         ssl.setKeyStorePassword("password");
 

